### ISSUE TITLE
Fix Build Failure in `unionlabs` by Pinning `uint` Dependency

### ISSUE THREAD

**Issue by srdtrk** [unionlabs/union#3143]
### **Describe the bug**
The `unionlabs` package encounters build failures when used as an external library due to incompatibility with the `v0.10.0` version of the `uint` crate. (`Cargo.lock` is ignored when a crate is imported as an external library). The root cause is that the package imports `uint = "*"`, which pulls in an incompatible version when used as a library. Reproducing the issue in this repo and fixing the dependency resolves the errors, as demonstrated in this repository. I have a working branch that addresses this issue and would love to open a PR to fix it.

### **To Reproduce**
1. Run `cargo update` (since the current `Cargo.lock` is ignored when building external crates).
2. Execute `cargo build -p unionlabs`.
3. Observe the error.

### **Expected behavior**
Package to build

### **Screenshots**
![image](https://github.com/user-attachments/assets/17984eab-fba1-47d2-a6ae-8b64be4ccf6a)

### **Fix:**
The error is resolved by explicitly selecting the `0.9` version of the `uint` crate in `lib/unionlabs/Cargo.toml`:
```diff
-uint                     = "*"
+uint                     = "0.9"
```

**Comment by srdtrk** [comment]
I realize that I tagged this as a site bug by mistake, this should be an app bug.

**Comment by benluelo** [comment]
good catch! this should be fixable downstream in your own crate(s) by pinning the `uint` dependency to 0.9 in your own Cargo.toml, as a quick fix until we get this patched here.

**Comment by benluelo** [comment]
fixed in #3152
